src/components/SignUpDialogueBox.tsx

```diff
@@ -180,68 +146,89 @@ const SignUpDialogueBox = (props: DialogueProps) => {
   return (
     <form
       className="bg-blue-500 flex flex-col h-full px-4 pt-4"
-      onSubmit={handleSubmit}>
+      onSubmit={handleSubmit(onSubmit)}>
       {/* Header */}
       <p className="text-center text-white font-semibold text-xl pb-4">
         Sign Up!
       </p>
-
       {/* Text Fields */}
       <input
-        className="text-center text-white bg-blue-800 focus:outline-none focus:shadow-outline border border-blue-500 mb-2"
+        name="username"
+        ref={register({ required: true })}
         placeholder="Enter Username"
-        value={username}
-        onChange={onChangeUsername}
+        className="text-center text-white bg-blue-800 focus:outline-none focus:shadow-outline border border-blue-500 mb-2"
       />
+      {errors.username && (
```

Comment:
i would destructure all these objects.  I will rarely reference something as `x.y` I usually destructure all variables at the top of the function